services/common/src/main/java/org/finra/gatekeeper/common/authfilter/parser/GatekeeperUserProfile.java

```diff
@@ -18,30 +18,64 @@
 package org.finra.gatekeeper.common.authfilter.parser;
 
 import java.util.Objects;
+import java.util.Set;
 
 public class GatekeeperUserProfile implements IGatekeeperUserProfile {
```

Comment:
The changes made to this are to cater to the header-based approach. I would define a different GatekeeperUserProfile that implements the IGatekeeperUserProfile in addition to a new interface that has the changes you want to introduce